deploy/test/test_cases/TEST_ID_21_helm_service_account_exists.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+set -euxo pipefail
+
+# This test verifies that when the user declares that they will provide their own Service Account and that Service Account exists
+# the Secrets Provider will deploy and complete successfully
+# TODO verify Job start up and completed logs
+
+setup_helm_environment
+
+create_k8s_role "another-"
+
+pushd ../../../
+  # RBAC will not be created by the Secrets Provider Helm Chart
+  export CREATE_SERVICE_ACCOUNT="false"
+
+  # These roles should not be created because of the above configuration
+  export SERVICE_ACCOUNT="another-secrets-provider-service-account"
+
+  fill_helm_chart
+  helm install -f "secrets-provider/ci/test-values.yaml" \
+    secrets-provider ./secrets-provider \
+    --set-file environment.conjur.sslCertificate.value="deploy/test/test_cases/conjur.pem"
```

Comment:
if i understand correctly, you create the `another-` resources and tell `helm` not to create its own resources to use yours. but can't we do it with the original resources? they are cleaned up between tests so if instead of creating `another-` resources we will create the original ones and verify the secrets-provider still works.

I am trying to understand whether we _really_ need all of the `another` resources because they add a huge complexity to all of these files. this is the main issue i have with this PR (as you may have noticed)

Reply:
In short, we create `another` resource and tell Helm _not_ create the new resources because I have already created them (depicted by `CREATE_SERVICE_ACCOUNT="false"` meaning "do not create rbac because I will supply them)

Those resources are `another-secrets-provider-service-account` `another-secrets-provider-role` `another-secrets-provider-role-binding` etc.

A couple of notes:

> they are cleaned up between tests so if instead of creating another- resources we will create the original ones and verify the secrets-provider still works.

I wanted to be very straightforward with the way tests are written. Additionally, in previous tests, resources are created via Helm and I wanted to be very purposeful with how I create them because ideally, the customer will not use Helm to create their own rbac resources and then us our chart.

I thought about using original resources but if you look in `fill_helm_chart` we have defaults so I wanted to use "another" to be completely sure that the mechanism works as expected and actually takes resources that I created previously and not those defaults.